volatility3/framework/symbols/windows/extensions/callbacks.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+from volatility3.framework import exceptions, objects
+from volatility3.framework.symbols.windows.extensions import pool
+
+vollog = logging.getLogger(__name__)
+
+
+class _SHUTDOWN_PACKET(objects.StructType, pool.ExecutiveObject):
+    """Class for _SHUTDOWN_PACKET objects found in IoSh pools.
+
+    This class serves as a base class for all pooled shutdown callback packets.
+
+    It exposes a function which sanity-checks structure members.
+    """
+
+    def is_valid(self) -> bool:
+        """
+        Perform some checks.
+        """
+        try:
+            if not (
+                self.Entry.Flink.is_readable()
+                and self.Entry.Blink.is_readable()
+                and self.DeviceObject.is_readable()
+            ):
+                return False
+
+            device = self.DeviceObject.dereference()
```

Comment:
Again, dereference() *shouldn't* need to be explicitly specified.  I'd prefer not to muddy the code with them when they're unnecessary in case people start picking up bad habits and assuming they're necessary.  If it *is* necessary, I'd like to know why in case we can improve the framework...